0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+
+The `urls` field of the [Get Ledger Metadata][] method returns a list of paths to other API methods. Each member of the `urls` field describes one path, which MUST be an HTTP(S)-formatted URL unless otherwise specified. Some paths contain [RFC6570][]-formatted variable sections in curly braces. The `urls` field MUST include all of the following:
+
+| Path name              | Variables     | HTTP Method | API Method(s)         |
+|:-----------------------|:--------------|:------------|:----------------------|
+| `transfers`            | None          | POST        | [Prepare Transfer][]   |
+| `websocket`            | None          | N/A path    | [WebSocket][] (MUST be a `ws://` or `wss://` URL.) |
+| `transfer`             | `{client_id}` | GET         | [Get Transfer][]      |
+| `transfer_rejection`   | `{client_id}` | PUT         | [Reject Transfer][]   |
+| `transfer_fulfillment` | `{client_id}` | GET, PUT    | [Get Transfer Fulfillment][], [Submit Fulfillment][] |
+| `account`              | `{name}`      | GET         | [Get Account][]       |
+
+The `urls` field of the metadata MAY also contain other methods provided by the ledger.
+
+Some resources also contain fields whose values are URLs pointing to other methods. These URLs MUST NOT contain [RFC6570][]-formatted variable sections. The following table maps URL fields to the API methods that can be accessed at those paths:
+
+| Resource | Field                | HTTP Method | API Method                   |
+|:---------|:---------------------|:------------|:-----------------------------|
+| Transfer | `id`                 | GET         | [Get Transfer][]             |
+| Transfer | `ledger`             | GET         | [Get Ledger Metadata][]      |
+| Transfer | `fulfillment`        | PUT         | [Submit Fulfillment][]       |
+| Transfer | `fulfillment`        | GET         | [Get Transfer Fulfillment][] |
+| Transfer | `transfer_rejection` | PUT         | [Reject Transfer][]          |
+| Transfer | `debit_account`      | GET         | [Get Account][]              |
+| Transfer | `credit_account`     | GET         | [Get Account][]              |
+| Message  | `from`               | GET         | [Get Account][]              |
+| Message  | `to`                 | GET         | [Get Account][]              |
+| Message  | `ledger`             | GET         | [Get Ledger Metadata][]      |
+
+
+### Date-Time
+[Date-Time]: #date-time
+
+All dates and times should be expressed in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date-time format with precision to the second or millisecond. The time zone offset should always be `Z`, for no offset. In other words, the date-time should be a **string** matching the one of the following formats:
+
+| Precision   | Format                     |
+|:------------|:---------------------------|
+| Second      | `YYYY-MM-DDTHH:mm:ss.sssZ` |
+| Millisecond | `YYYY-MM-DDTHH:mm:ssZ`     |
```

Comment:
I'd rather make milliseconds optional because it saves some implementers the trivial concern of whether their programming language's ISO-String function provides milliseconds by default.